pkg/login/social/common.go

```diff
@@ -54,21 +60,20 @@ func (s *SocialBase) httpGet(client *http.Client, url string) (response httpGetR
 		}
 	}()
 
-	body, err := io.ReadAll(r.Body)
-	if err != nil {
-		return
+	body, errRead := io.ReadAll(r.Body)
+	if errRead != nil {
+		return nil, errRead
 	}
 
-	response = httpGetResponse{body, r.Header}
+	response := &httpGetResponse{body, r.Header}
 
 	if r.StatusCode >= 300 {
-		err = fmt.Errorf(string(response.Body))
-		return
+		return nil, fmt.Errorf("invalid response status code %d: %s", r.StatusCode, string(response.Body))
```

Comment:
super nit: to me "invalid response status code" reads slightly weird. Maybe
```suggestion
		return nil, fmt.Errorf("unsuccessful response status code %d: %s", r.StatusCode, string(response.Body))
```